superset/dashboards/filters.py

```diff
@@ -23,6 +24,20 @@
 from superset.views.base import BaseFilter, get_user_roles
 
 
+class DashboardTitleOrSlugFilter(BaseFilter):  # pylint: disable=too-few-public-methods
+    name = _("Title or Slug")
+    arg_name = "title_or_slug"
+
+    def apply(self, query, value):
+        ilike_value = f"%{value}%"
```

Comment:
we may want to handle `None` or `''` cases here and not apply the filter. I think running an `ilike %%` can be very slow on some dbs. I can handle these cases before issuing the request, however we may want some protection via the api.

Reply:
Good point!